common/plugins/eu.esdihumboldt.hale.common.instance/src/eu/esdihumboldt/hale/common/instance/processing/InstanceProcessor.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2017 wetransform GmbH
+ * 
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     wetransform GmbH <http://www.wetransform.to>
+ */
+
+package eu.esdihumboldt.hale.common.instance.processing;
+
+import eu.esdihumboldt.hale.common.core.service.ServiceProvider;
+import eu.esdihumboldt.hale.common.instance.model.Instance;
+import eu.esdihumboldt.hale.common.instance.model.InstanceReference;
+
+/**
+ * {@link Instance} processor
+ * 
+ * @author Florian Esser
+ */
+public interface InstanceProcessor {
+
+	/**
+	 * Set the context service provider.
+	 * 
+	 * @param services the service provider
+	 */
+	void setServiceProvider(ServiceProvider services);
+
+	/**
+	 * Processing before an instance is stored.
+	 * 
+	 * @param instance the instance
+	 */
+	void beforeStore(Instance instance);
```

Comment:
My idea was to provide hooks for 
- instance manipulation before storage in OrientDB (currently also handled by `StoreInstancesJob.processInstances` which I thought could maybe be replaced)
- making use of the `InstanceReference` after storage in OrientDB
and needing only one implementation to cover both.

I agree that this is tightly coupled to the StoreInstancesJob context. If we want to make this more generic and extendable, we could – instead of having before/after methods – add execution phases like e.g. `ON_LOAD` (for index population) or `AFTER_TYPE_TRANSFORMATION` (for filling collectors).